library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/StringObfuscatorTest.kt

```diff
@@ -0,0 +1,145 @@
+                fakeExpectedChunk3 +
+                forge.aString(fakeExpectedChunk4.length) { forge.anAlphaNumericalChar() } +
+                fakeExpectedChunk5
+            ).times(n)
+
+        // When
+        val obfuscatedText = testedObfuscator.obfuscate(fakeText)
+
+        // Then
+        assertThat(obfuscatedText).isEqualTo(fakeExpectedText)
+    }
+
+    @Test
+    fun `M mask String W maskString(){string with carriage return character}`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeExpectedChunk1 = forge.aString(size = forge.anInt(1, max = 10)) { '\r' }
+        val fakeExpectedChunk2 = forge.aString { 'x' }
+        val fakeExpectedChunk3 = forge.aString(size = forge.anInt(1, max = 10)) { '\r' }
+        val fakeExpectedChunk4 = forge.aString { 'x' }
+        val fakeExpectedChunk5 = forge.aString(size = forge.anInt(1, max = 10)) { '\r' }
+        val n = forge.anInt(min = 1, max = 10)
+        val fakeExpectedText = (
+            fakeExpectedChunk1 +
+                fakeExpectedChunk2 +
+                fakeExpectedChunk3 +
+                fakeExpectedChunk4 +
+                fakeExpectedChunk5
+            ).times(n)
+        val fakeText = (
+            fakeExpectedChunk1 +
+                forge.aString(fakeExpectedChunk2.length) { forge.anAlphaNumericalChar() } +
+                fakeExpectedChunk3 +
+                forge.aString(fakeExpectedChunk4.length) { forge.anAlphaNumericalChar() } +
+                fakeExpectedChunk5
+            ).times(n)
+
+        // When
+        val obfuscatedText = testedObfuscator.obfuscate(fakeText)
+
+        // Then
+        assertThat(obfuscatedText).isEqualTo(fakeExpectedText)
+    }
+
+    @Test
+    fun `M mask String W maskString(){string with whitespace character}`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeExpectedChunk1 = forge.aWhitespaceString()
+        val fakeExpectedChunk2 = forge.aString { 'x' }
+        val fakeExpectedChunk3 = forge.aWhitespaceString()
+        val fakeExpectedChunk4 = forge.aString { 'x' }
+        val fakeExpectedChunk5 = forge.aWhitespaceString()
+        val n = forge.anInt(min = 1, max = 10)
+        val fakeExpectedText = (
+            fakeExpectedChunk1 +
+                fakeExpectedChunk2 +
+                fakeExpectedChunk3 +
+                fakeExpectedChunk4 +
+                fakeExpectedChunk5
+            ).times(n)
+        val fakeText = (
+            fakeExpectedChunk1 +
+                forge.aString(fakeExpectedChunk2.length) { forge.anAlphaNumericalChar() } +
+                fakeExpectedChunk3 +
+                forge.aString(fakeExpectedChunk4.length) { forge.anAlphaNumericalChar() } +
+                fakeExpectedChunk5
+            ).times(n)
+
+        // When
+        val obfuscatedText = testedObfuscator.obfuscate(fakeText)
+
+        // Then
+        assertThat(obfuscatedText).isEqualTo(fakeExpectedText)
+    }
+
+    private fun String.times(n: Int): String {
```

Comment:
it seems the default [repeat](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/repeat.html) should do the job?